### ISSUE TITLE
@typescript-eslint/no-unused-vars - `{#each}` block

### ISSUE THREAD

**Issue by tmarsik42** [sveltejs/eslint-plugin-svelte#469]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.40.0

### What version of `eslint-plugin-svelte` are you using?

2.27.4

### What did you do?

<details>
<summary>Configuration</summary>

```
module.exports = {
    root: true,
    parser: "@typescript-eslint/parser",
    parserOptions: {
        sourceType: "module",
        ecmaVersion: 2020,
        project: "./tsconfig.json",
        extraFileExtensions: [".svelte"]
    },
    plugins: ["@typescript-eslint"],
    extends: ["plugin:svelte/prettier", "prettier"],
    ignorePatterns: ["*.cjs"],
    overrides: [
        {
            files: ["*.svelte"],
            parser: "svelte-eslint-parser",
            parserOptions: {
                parser: "@typescript-eslint/parser"
            },
            rules: {
                "no-magic-numbers": 0, //vytváří hodně errorů v parametrech komponent (např. <Column/>, počet sloupců),
                "max-lines": 0, // markup je hodně verbózní
                "a11y-click-events-have-key-events": 0,
                "no-console": [
                    process.env.NODE_ENV === "production" ? "error" : "warn",
                    { allow: ["error", "warn"] }
                ]
            }
        },
        {
            files: ["*.ts"],
            rules: {
                "no-magic-numbers": ["error", { ignore: [0, 1, -1] }]
            }
        }
    ],
    rules: {
        "a11y-click-events-have-key-events": 0
    },
    env: {
        browser: true,
        es2020: true,
        node: true
    }
};

```
</details>

```svelte
 {#each Array(5) as _}
            <div class="user">
                <span style="padding: var(--cds-spacing-04);">
                    <SkeletonPlaceholder />
                </span>
                <h6 style:padding="{0}">
                    <SkeletonText width="10rem" />
                </h6>
            </div>
{/each}
```

### What did you expect to happen?

Not throwing no unused vars on `_` when not used

### What actually happened?

no unused vars

### Link to **GitHub Repo** with Minimal Reproducible Example

no repro
<img width="785" alt="Snímek obrazovky 2023-05-09 v 15 10 16" src="https://github.com/sveltejs/eslint-plugin-svelte/assets/95856146/55e95d08-6499-4dbd-b8a9-9811c3f3c1a4">
